File: api_platform/api-platorm-backend/admin/response/api_category_response.py
from common.response.response_body import BaseResponseBody,Tree
from common.id_generator.id_util import SnowFlakeID

class APICategoryResponse(BaseResponseBody):
    parent_id:SnowFlakeID
    category_name:str
    sort:int

class APICategoryTree(Tree["APICategoryTree"]):
    category_name:str
    sort:int

APICategoryResponse.model_rebuild()
APICategoryTree.model_rebuild()

File: api_platform/api-platorm-backend/user/clientuser/src/main/java/com/whs/apiplatform/api/response/ApiCategoryResponse.java
package com.whs.apiplatform.api.response;

import com.whs.apiplatform.common.tree.TreeNode;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
public class ApiCategoryResponse implements TreeNode<ApiCategoryResponse> {
    private Long id;
    private Long parentId;
    private String categoryName;
    private Integer sort;
    private String ancestors;
    private Long creatorId;
    private Date createTime;
    private Date updateTime;

    private List<ApiCategoryResponse> children;
}
